packages/fuel-indexer/src/executor.rs

```diff
@@ -17,10 +17,301 @@ use wasmer::{
+                                        TransactionStatus::Submitted {
+                                            submitted_at: Utc
+                                                .timestamp_opt(submitted_at.to_unix(), 0)
+                                                .single()
+                                                .expect(
+                                                    "Failed to parse transaction timestamp"
+                                                ),
+                                        }
+                                    }
+                                    GqlTransactionStatus::SqueezedOut { reason } => {
+                                        TransactionStatus::SqueezedOut { reason }
+                                    }
+                                };
+
+                                let tx_data = TransactionData {
+                                    receipts,
+                                    status,
+                                    transaction,
+                                    id: TxId::from(trans.id),
+                                };
+                                transactions.push(tx_data);
+                            }
+                        }
+                        Err(e) => {
+                            error!(
+                                "Client communication error fetching transactions: {:?}",
+                                e
+                            )
+                        }
+                    };
+                }
+
+                let block = BlockData {
+                    height: block.header.height.0,
+                    id: Bytes32::from(block.id),
+                    producer,
+                    time: block.header.time.0.to_unix(),
+                    transactions,
+                };
+
+                block_info.push(block);
+            }
+
+            let result = executor.handle_events(block_info).await;
+
+            if let Err(e) = result {
+                error!("Indexer executor failed {e:?}, retrying.");
+                sleep(Duration::from_secs(DELAY_FOR_SERVICE_ERR)).await;
+                retry_count += 1;
+                if retry_count < INDEX_FAILED_CALLS {
+                    continue;
+                } else {
+                    error!("Indexer failed after retries, giving up. <('.')>");
+                    break;
+                }
+            }
+
+            if cursor.is_none() {
+                info!("No new blocks to process, sleeping.");
+                sleep(Duration::from_secs(DELAY_FOR_EMPTY_PAGE)).await;
+
+                empty_block_reqs += 1;
+
+                if empty_block_reqs == MAX_EMPTY_BLOCK_REQUESTS {
+                    error!("No blocks being produced, giving up. <('.')>");
+                    break;
+                }
+            } else {
+                next_cursor = cursor;
+            }
```

Comment:
I think that we need to reset `empty_block_reqs` in the else block here. Otherwise, we keep incrementing with the old value even if a new block has successfully been received.

Reply:
@deekerno Doh! Good find (I originally thought you were talking about something else). Fix incoming